semeio/workflows/localisation/local_script_lib.py

```diff
@@ -457,61 +829,42 @@ def initialize(cls):
         cls.corr_name = None
 
     @classmethod
-    def write(cls, node_name, corr_spec, grid_for_field, log_level):
-        corr_name = corr_spec.name
-        ref_point = corr_spec.ref_point
-        grid = grid_for_field
-        if corr_spec.field_scale is None:
-            raise ValueError(
-                "Scaling values for 3D parameters require "
-                f"specification of keyword 'field_scale' for {cls.corr_name}"
-            )
-        method_name = corr_spec.field_scale.method
-
-        if method_name == "gaussian_decay":
-            decay_obj = GaussianDecay(
-                ref_point,
-                corr_spec.field_scale.main_range,
-                corr_spec.field_scale.perp_range,
-                corr_spec.field_scale.azimuth,
-                grid,
-            )
-            scaling_values = cls._calculate_scaling_param(grid, decay_obj)
-        elif method_name == "exponential_decay":
-            decay_obj = ExponentialDecay(
-                ref_point,
-                corr_spec.field_scale.main_range,
-                corr_spec.field_scale.perp_range,
-                corr_spec.field_scale.azimuth,
-                grid,
-            )
-            scaling_values = cls._calculate_scaling_param(grid, decay_obj)
-        else:
-            raise KeyError(f" Method name: {method_name} is not implemented")
+    def write_qc_parameter(
+        cls,
+        node_name,
+        corr_name,
+        field_scale,
+        grid,
+        param_for_field,
+        log_level=LogLevel.OFF,
+    ):
+        if param_for_field is None or field_scale is None:
+            return
+
+        scaling_values = np.reshape(
+            param_for_field, (grid.getNX(), grid.getNY(), grid.getNZ()), "F"
+        )
 
         # Write scaling parameter  once per corr_name
         if corr_name != cls.corr_name:
             cls.corr_name = corr_name
+            # Need a parameter name <= 8 character long
             scaling_kw_name = "S_" + str(cls.scaling_param_number)
             scaling_kw = grid.create_kw(scaling_values, scaling_kw_name, False)
             filename = (
                 cls.corr_name + "_" + node_name + "_" + scaling_kw_name + ".GRDECL"
             )
-            debug_print(f"Write file: {filename}", LogLevel.LEVEL3, log_level)
+            print(
```

Comment:
I didn't understand what to do here. The messages are different in my code and more detailed than it was before. Now it is:

> print("Write calculated scaling factor  with name: "
>            f"{scaling_kw_name} to file: {filename}"
>         )

and 

> debug_print(
>                 f"Write calculated scaling factor with name: "
>                 f"{scaling_kw_name} to file: {filename}",
>                 LogLevel.LEVEL3,
>                 log_level,
>             )

Reply:
My question was if you needed both `print` and `debug_print`?

Reply:
It is OK to have most on log file, but sometimes it is useful to get some message on the terminal screen as well. In this case it helps to report that the QC files for scaling factor is created since it is not 'hidden' in the log file that is not so easy to find.